olive/evaluator/olive_evaluator.py

```diff
@@ -1106,6 +1141,12 @@ def validate_metrics(cls, v):
         expected_rank_set = set(range(1, len(sub_type_with_rank) + 1))
         # Check if all ranks are present
         if rank_set != expected_rank_set:
-            raise ValueError(f"Priorities must be unique and in the range 1 to {metric_len}")
+            raise ValueError(
+                f"Priorities must be unique and in the range 1 to {metric_len}\n"
+                f"\tActual: {rank_set}\n\tExpected: {expected_rank_set}"
+            )
 
         return v
+
+    def create_evaluator(self, model: "OliveModelHandler" = None) -> OliveEvaluator:
+        return Registry.get(self.type or str(model.framework))(**(self.type_args or {}))
```

Comment:
qq: In which case we will use type to get an evaluator?

Reply:
I am introducing custom evaluators to support internal i.e. MS specific evaluation techniques. Since these are internal, they cannot be part of public Olive repo.